crates/oxc_module_lexer/src/lib.rs

```diff
@@ -164,10 +171,37 @@ impl<'a> Visit<'a> for ModuleLexer<'a> {
             se: expr.span.end,
             d: ImportType::DynamicImport(expr.span.start + 6),
             a: expr.arguments.first().map(|e| e.span().start),
+            t: false,
         });
         walk_import_expression(self, expr);
     }
 
+    fn visit_ts_import_type(&mut self, impt: &TSImportType<'a>) {
+        let (source, source_span_start, source_span_end) =
+            if let TSType::TSLiteralType(literal) = &impt.parameter {
+                if let TSLiteral::StringLiteral(s) = &literal.literal {
+                    (Some(s.value.clone()), s.span.start, s.span.end)
+                } else {
+                    let span = literal.span();
+                    (None, span.start, span.end)
+                }
+            } else {
+                let span = impt.parameter.span();
+                (None, span.start, span.end)
```

Comment:
Is there a better way to avoid repeating code here?

Reply:
This is fine. What is being repeated?